graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/SearchRequestFactory.java

```diff
@@ -43,14 +45,23 @@ public class SearchRequestFactory {
     private final SortOrderMapper sortOrderMapper;
     private final boolean allowHighlighting;
     private final boolean allowLeadingWildcardSearches;
+    private final UsedSearchFiltersToQueryStringsMapper usedSearchFiltersToQueryStringsMapper;
 
     @Inject
     public SearchRequestFactory(SortOrderMapper sortOrderMapper,
                                 @Named("allow_highlighting") boolean allowHighlighting,
-                                @Named("allow_leading_wildcard_searches") boolean allowLeadingWildcardSearches) {
+                                @Named("allow_leading_wildcard_searches") boolean allowLeadingWildcardSearches,
+                                UsedSearchFiltersToQueryStringsMapper usedSearchFiltersToQueryStringsMapper) {
         this.sortOrderMapper = sortOrderMapper;
         this.allowHighlighting = allowHighlighting;
         this.allowLeadingWildcardSearches = allowLeadingWildcardSearches;
+        this.usedSearchFiltersToQueryStringsMapper = usedSearchFiltersToQueryStringsMapper;
+    }
+
+    public SearchRequestFactory(SortOrderMapper sortOrderMapper,
+                                boolean allowHighlighting,
+                                boolean allowLeadingWildcardSearches) {
+        this(sortOrderMapper, allowHighlighting, allowLeadingWildcardSearches, new IgnoreSearchFilters());
     }
```

Comment:
Thanks for asking about this one. How concerned should we be with backwards-compatibility/non-breaking changes to the search API code? The only reason for keeping this constructor signature (was present before this PR) and adding the explicit `IgnoreSearchFilters` instance was to maintain backwards compatibility in case somehow had manually called the constructor or extended this class.

Reply:
I usually mention API changes like that in UPGRADING.md.
You could ask @bernd if he accepts this change, to be on the safe side.

Reply:
@bernd As part of this PR, some changes to open source search classes were needed. The changes are mostly adding an additional method parameter `List<UsedSearchFilter> filters` and also adding an additional constructor parameter (see above change for this comment).

In the first version of the PR (as it's currently coded), I had tried to avoid API changes by adding methods and constructors. Is that necessary, or should I make changes to the original methods/constructors, and document the changes in `UPGRADING.md`?

Reply:
@bernd and I talked, and the backwards-compatibility changes are not needed. This area is really a special use case, and is not likely to be depended on by 3rd party code. I'll remove the extra code for backwards compatibility in a commit shortly.